client/src/components/company-input/company-input.tsx

```diff
@@ -20,18 +21,38 @@ export interface CompanyInputFormProps {
   onFileUpload: (file: CustomFile) => void;
 }
 
+interface valueProps {
+  submissionId: string;
+  participationCode: string;
+  industry: string;
+  size: string;
+}
+
 export const CompanyInputForm: FC<CompanyInputFormProps> = (props) => {
   const urlParams = new URLSearchParams(window.location.search);
   const session_id = urlParams.get('session_id') || '';
   const participant_token = urlParams.get('participant_code') || '';
 
+  const context = useContext(SessionContext);
+  const { setSessionId, setParticipantCode, setIndustry, setCompanySize } = context;
   const [initialValues, setInitialValues] = useState({
     submissionId: session_id,
     participationCode: participant_token,
     industry: undefined,
     size: undefined
   });
 
+  const FormObserver: React.FC = () => {
```

Comment:
NIT: This could be its own component
if you have a prop that is a function called onChange that you subscribe to on the component itself

<FormObserver onChange={(values) => {
  // update context
}}/>

This probably doesn't need to change but just some feedback